src/main/java/org/apache/commons/cli/Util.java

```diff
@@ -17,16 +17,45 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
 package org.apache.commons.cli;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
 /**
  * Contains useful helper methods for classes within this package.
  */
-final class Util {
+public final class Util {
```

Comment:
I think we should keep as much of this internal stuff hidden. If you need _new_ shared utilities then let's put them close to where they're used, in the new package, in its own package-private class. Since the PR reuses some code from here, then we have to make the class public, but I would _really_ like to minimize the public footprint of this utility stuff to give us maximum flexibility going forward.

Reply:
Either way, since this is a new "block" of code, it should show up clearly in a PR. We can do it in two steps (PRs). If done in 2 steps, we'll have to remember to reduce the visibility of the new util methods once they are in the help package because it might be debatable what kind of utils implementations will need or want to use/reuse. I just want to be careful as to how much we are increasing the public (including protected) footprint. At some point, an implementation might just choose to use Commons Lang and Commons Text.